src/main/java/xbot/common/logic/Latch.java

```diff
@@ -19,6 +22,15 @@ public Latch(boolean initialValue, EdgeType latchType) {
         this.latchType = latchType;
     }
 
+    public Latch(boolean initialValue, EdgeType latchType, Consumer<EdgeType> callback) {
```

Comment:
The latch changes are to simplify usage. Whereas before you had to supply an `Observer`, which was passed a generally useless "sender" and an `Object` you had to cast for use, now you can pass your callback to the constructor (or add it later as before) without having to do the cast.

Reply:
This is a good improvement!